userspace/libscap/scap_savefile.c

```diff
@@ -1120,58 +1211,92 @@ static int32_t scap_read_machine_info(scap_t *handle, gzFile f, uint32_t block_l
 static int32_t scap_read_proclist(scap_t *handle, gzFile f, uint32_t block_length, uint32_t block_type)
 {
 	size_t readsize;
+	size_t subreadsize = 0;
 	size_t totreadsize = 0;
 	size_t padding_len;
 	struct scap_threadinfo tinfo;
 	uint16_t stlen;
 	uint32_t padding;
 	int32_t uth_status = SCAP_SUCCESS;
 	struct scap_threadinfo *ntinfo;
-
-	tinfo.fdlist = NULL;
-	tinfo.flags = 0;
-	tinfo.vmsize_kb = 0;
-	tinfo.vmrss_kb = 0;
-	tinfo.vmswap_kb = 0;
-	tinfo.pfmajor = 0;
-	tinfo.pfminor = 0;
-	tinfo.env_len = 0;
-	tinfo.vtid = -1;
-	tinfo.vpid = -1;
-	tinfo.cgroups_len = 0;
-	tinfo.filtered_out = 0;
-	tinfo.root[0] = 0;
-	tinfo.sid = -1;
-	tinfo.vpgid = -1;
-	tinfo.clone_ts = 0;
-	tinfo.tty = 0;
-	tinfo.exepath[0] = 0;
+	uint32_t toread;
+	int fseekres;
 
 	while(((int32_t)block_length - (int32_t)totreadsize) >= 4)
 	{
+		tinfo.fdlist = NULL;

```

Comment:
Now that tinfo is being initialized in every loop iteration, why not define it inside the loop as well? I'd suggest doing the same for ntinfo as well, as it seems to only be used on lines 1695-1711.

Reply:
done